### ISSUE TITLE
Plain zero (0) without a dot or slash 

### ISSUE THREAD

**Issue by jashson** [JetBrains/JetBrainsMono#625]
When using this font on a website in a context, where only numbers are used, it's not easy to get the difference between the 0 and the 8. No matter if it is displayed as dotted zero or slashed zero. 

It would be great to have another option via css like:
`font-feature-settings: "zero-plain"` for having a zero without a dot or slash

**Comment by YVVT** [comment]
It may be difficult to distinguish from capital letter O

**Comment by jashson** [comment]
It’s only a problem when you mix text and numbers. But if you in a context (i.e. table row with just number values) where only numbers are displayed, it won’t be an issue und will increase readability.

**Comment by BD5** [comment]
I'm here for the same reason, I'm having a hard time telling the difference between `0` and `8` at smaller point sizes. However, my request is kind of the opposite. I'd like an alternate `8` where the top circle is smaller, both in height and width, than the bottom circle. I'll also suggest a zero with a  dot that is thinner and a bit taller (assuming the dot can be made thinner), or possibly drop the current dot lower instead of centered, which should make it that much easier to distinguish `0` with a new `8`.

And as I type this, I wonder if maybe ligatures for `08` and `80` could be created, though I have no idea what such ligatures would look like. (Yep, just thinking out loud here.)

**Comment by gavinbaradic** [comment]
+1

**Comment by Unknown78** [comment]
I agree; this issue is particularly evident in embedded or bare-metal programming.
Hexadecimal strings, such as the memory address `0x08080088`, are incredibly difficult to read accurately.
Therefore making the font ironically illegible.

---

Based on https://www.programmingfonts.org

I've looked the font that used non-slashed or non-dotted zero by default:
<details><summary><b>(Click to expand/shrink)</b></summary>

```
APL385
Aurulent Sans Mono
Average Mono
Azeret Mono
B612 Mono
Bedstead
Chivo Mono
Computer Modern Unicode Typewriter 
Courier Prime
Cutive Mono
Droid Sans
Effects Eighty
GNU Freefont
IBM Courier
Latin Modern Mono
Lekton
Luxi Mono
Nimbus Mono PS
NK57 Monospace
Paper Mono
Press Start 2P
saxMono
Sk-Modernist
TeX Gyre Cursor
VT323
```
</details>

But the trade-off of those fonts is that the number `0` and the letter `o` is illegible.

Now, there is one font outside that list that I think is the most unique in this case, named [Bedstead](https://bjh21.me.uk/bedstead/). 
It replaces zero with quasi-diamond, therefore making it legible to the number of 8 and to the letter of O. The name of the font is 
But the trade-off is that the curly-braces are not legible enough from the parenthesis.

<img width="777" height="270" alt="Image" src="https://github.com/user-attachments/assets/923175d5-4721-4660-b36a-a03bb137c264" />

---

At this point, I might use the [Iosevka Customizer ](https://typeof.net/Iosevka/customizer) instead, lol.

Iosevka also has option to set the zero into quasi-diamond. And I"m glad someone has already made the automation to also include the nerd patch, https://github.com/awnion/custom-iosevka-nerd-font/tree/main#build-your-own

---

I think what I really wanted is that for JetBrains Mono to have an option for the zero to be filled in diamond shaped.


That's even not enough, the number of eight needs to be also legible to capital B.
```
    eight = "crossing-asymmetric"
    zero = "diamond-broken-vertical-bar"
    capital-b = "more-asymmetric-interrupted-bilateral-serifed"
```

<img width="223" height="61" alt="Image" src="https://github.com/user-attachments/assets/95e620d7-b930-4c52-8fca-768822228e21" />

I've find that `diamond-broken-vertical-bar` from Iosevka Customizer is the closest one, although I prefer it to be completely filled.

---

Update:
Although the `zero = "diamond-broken-vertical-bar"` increased legibility, it also hurts readability, as it is so jarring.
There's always trade-off.
I'm gonna try `zero = "diamond-vertical-bar-cutout"`
